docs/tre-workspace-authors/firewall-rules.md

```diff
@@ -11,118 +8,188 @@ A TRE service may require certain firewall rules to be opened in the TRE firewal
 
 Please be aware when opening firewall rules there is the potential for data to be leaked from the workspace to the external location.
 
-## Using Terraform to open firewall rules
-
-Until a mechanism to update shared services has been implemented, firewall rule updates should be done using terraform as part of the service deployment. The aim is to create a firewall rule that grants access from the workspace's address space to the external location. A challenge with this is that the rule must use a priority that has not been used by any other rule.
-
-1. Create a `firewall.tf` file in the `terraform` directory of the workspace.
-
-1. Add the following code to the `firewall.tf` file to enable the TRE firewall and workspace network to be referenced:
-
-    ```terraform
-    data "azurerm_firewall" "fw" {
-        name                = "fw-${var.tre_id}"
-        resource_group_name = "rg-${var.tre_id}"
-    }
-
-    data "azurerm_virtual_network" "ws" {
-        name                = "vnet-${var.tre_id}-ws-${var.workspace_id}"
-        resource_group_name = data.azurerm_resource_group.ws.name
-    }
-    ```
-
-1. Define a local variable that contains the locations that access should be allowed to, and the naming format for the service resources for example:
-
-    ```terraform
-    locals {
-        allowed_urls                     = ["*.anaconda.com", "*.anaconda.org"]
-        service_resource_name_suffix    = "${var.tre_id}-ws-${var.workspace_id}-svc-${local.service_id}"
-    }
-    ```
-
-1. Log into the Azure CLI using service principal details:
-
-    ```terraform
-    resource "null_resource" "az_login" {
-        provisioner "local-exec" {
-            command = "az login --identity -u '${var.arm_client_id}'"
-        }
-
-        triggers = {
-            timestamp = timestamp()
-        }
-    }
-    ```
-
-1. Call the `get_firewall_priorities.sh` script to find the next available priority:
-
-    ```terraform
-    data "external" "rule_priorities" {
-        program = ["bash", "-c", "./get_firewall_priorities.sh"]
+## Firewall Rules in Template Schema
+
+Azure TRE uses the `template_schema.json` file of the service in question (e.g. `templates/workspace_services/azureml/template_schema.json`) to define firewall rules. These rules are configured in the `pipeline` section of the end of the schema file.
```

Comment:
nit: "of the end of the schema file"...should that be "*at* the end of the schema file?"

Reply:
```suggestion
Azure TRE uses the `template_schema.json` file of the service in question (e.g. `templates/workspace_services/azureml/template_schema.json`) to define firewall rules. These rules are configured in the `pipeline` section at the end of the schema file.
```